src/zope/schema/interfaces.py

```diff
@@ -387,13 +535,23 @@ class IInt(IMinMax, IField):
         )
 
 
-class IFloat(IMinMax, IField):
-    """Field containing a Float."""
+class IFloat(IReal):
+    """
+    Field containing exactly the native class :class:`float`.
+
+    :class:`IReal` is a more general interface, allowing all of
+    floats, ints, and fractions.
+
+    .. seealso:: :class:`zope.schema.Float`
+    """
```

Comment:
`float` is a specific type of `Real`, so the field definition is already basically correct from an interface and documentation perspective.

It runs into the same issue as `IDecimal` though, the `_type` of `Float` restricts them to actual `float` values, regardless of what the schema says. We could be more specific with putting `Float` fields into the schema, but we run into bootstrap problems with that (which could probably be overcome by moving the definition of the `Float` class), but it's all better than it was before.

Reply:
OK.  Logically, an `IFloat` field *could* take an integer as the default / min / max.

Reply:
I agree that logically it could, and I actually tried to put that in a test (which failed). That's how I know that the `_type` constraint controls that and rejects them 😄 That's one area that `IReal` and `Real` are improvements.